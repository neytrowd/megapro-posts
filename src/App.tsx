import React from 'react';
import './assets/styles/index.scss';
import Page from './layouts/page';
import { ThemeProvider, CssBaseline } from '@mui/material';
import theme from './assets/theme';
import { Routes, Route } from 'react-router-dom';
import routes from './constants/routes';

function App() {
   return (
      <div className="App">
         <ThemeProvider theme={theme}>
            <CssBaseline />
            <Page title="Mega pro app">
               <Routes>
                  {routes.map((route) => (
                     <Route key={route.path} path={route.path} element={route.component} />
                  ))}
               </Routes>
            </Page>
         </ThemeProvider>
      </div>
   );
}

export default App;
