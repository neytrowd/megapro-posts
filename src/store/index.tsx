import { combineReducers, configureStore } from '@reduxjs/toolkit';
import postSlice from './reducers/postSlice';
import infoSlice from './reducers/infoSlice';
import userSlice from './reducers/userSlice';

const rootReducer = combineReducers({
   posts: postSlice,
   info: infoSlice,
   users: userSlice,
});

export const store = configureStore({
   reducer: rootReducer,
});

export type RootState = ReturnType<typeof rootReducer>;
export type AppDispatch = typeof store.dispatch;
