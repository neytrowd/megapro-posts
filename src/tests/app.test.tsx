import React from 'react';
import { screen } from '@testing-library/react';
import App from '../App';
import { store } from '../store';
import { renderWithRouterAndProvider } from './helpers/render';

describe('App component', () => {
   test('should be in the document', () => {
      renderWithRouterAndProvider(<App />);
      expect(screen.getByText(/Mega pro app/i)).toBeInTheDocument();
      screen.debug();
   });

   test('should match snapshot', () => {
      renderWithRouterAndProvider(<App />);
      expect(screen.getByText(/Mega pro app/i)).toMatchSnapshot();
   });
});
