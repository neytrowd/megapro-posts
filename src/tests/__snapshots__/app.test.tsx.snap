// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should match snapshot 1`] = `
<p
  class="MuiTypography-root MuiTypography-body1 title css-ahj2mt-MuiTypography-root"
>
  Mega pro app
</p>
`;
